### PR TITLE
fix: changed core profiler industry types to the correct format

### PR DESCRIPTION
**Pull request by rjchow**
- changed profiler API to match what we're actually storing, Industry[]
- changed some core profiler steps to use the API instead of saving options directly

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

While working on some improvements to the Core Profiler I noticed that the industry types were wrong, and it was not caught because we were directly writing to the options instead of using the API.

This PR changes the Core Profiler to use the API instead, which performs validation and also supports merging a partial update into the object.

Merging a partial update was something that was done previously by fetching in the frontend and merging in the frontend. This can result in stale contents, but this condition was not triggered as there was no further update to the object in the later steps. (I noticed it as I started writing new content in the mentioned work above)

Aside from that there should be no actual change to the content being persisted.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set up WooCommerce and go through the Core Profiler
2. The two pages that saw code changes are the User Profile and the Business Info pages. Check that the fields are correctly persisted - this can be done by either checking using WC Admin Test Helper or by completing the Core Profiler and revisiting it to see that the previously entered information is correctly populated. (With the exception of the data opt-in as that's being worked on in another PR)

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/client/core-profiler/index.tsx [hunk 1]
```diff
@@ -473,26 +469,23 @@ const updateBusinessInfo = fromPromise(
 			payload: BusinessInfoPayload;
 		};
 	} ) => {
-		const refreshedOnboardingProfile = ( await resolveSelect(
```
Comment (rjchow) [inline plugins/woocommerce/client/admin/client/core-profiler/index.tsx:476]: the API merges the fields into the existing object so no need to fetch & merge on the frontend. This was causing consistency issues due to stale values in the fetch

File: plugins/woocommerce/client/admin/client/core-profiler/index.tsx [hunk 2]
```diff
@@ -473,26 +469,23 @@ const updateBusinessInfo = fromPromise(
...
-					input.payload.storeEmailAddress.length > 0
-						? input.payload.storeEmailAddress
-						: null,
-			},
-		} );
+				...( input.payload.industry && {
+					industry: [ input.payload.industry ],
+				} ),
+				...( input.payload.storeEmailAddress.length > 0 && {
+					store_email: input.payload.storeEmailAddress,
```
Comment (ilyasfoo) [inline plugins/woocommerce/client/admin/client/core-profiler/index.tsx:485]: I realized that merchants can't remove email address by emptying the field. It'll still populated by the previously saved value. I'm unsure what's the intended behaviour, but I think there would be cases where merchant wants to empty the email field. What do you think?

**Review by ilyasfoo — COMMENTED** [review]
Looking great, but I have a minor comment/question:

File: plugins/woocommerce/client/admin/client/core-profiler/index.tsx (hunk 2 above)
Comment (rjchow) [reply plugins/woocommerce/client/admin/client/core-profiler/index.tsx:485]: that’s a very good catch, I’ll make the API nullable instead then

**Comment by rjchow** [comment]
Hey @ilyasfoo I've made it nullable now, and to omit the key if the value is same as the previous - I think this should work!

**Review by ilyasfoo — APPROVED** [review]
Thanks, @rjchow! Tests well 🚢

**Review by chihsuan — APPROVED** [review]
Looks good and tested well! 👍

**Pull request merged**